File: backend-dotnet/Models/Todo.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend_dotnet.Models
{
    [Table("todos")]
    public class Todo
    {
        [Column("id")]
        public int Id { get; set; }

        [Column("title")]
        public string Title { get; set; } = string.Empty;

        [Column("description")]
        public string? Description { get; set; }

        [Column("priority")]
        public string? Priority { get; set; }

        [Column("duedate")]
        public DateTime? DueDate 
        { 
        get => _dueDate;
        set => _dueDate = value?.ToUniversalTime();
        }
        private DateTime? _dueDate;

        [Column("completed")]
        public bool Completed { get; set; }
    }
}

File: frontend-angular/src/app/components/todo/todo.ts
import { Component } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { HttpClient } from '@angular/common/http';

@Component({
  selector: 'app-todo',
  standalone: true,
  imports: [CommonModule, FormsModule],
  templateUrl: './todo.html',
  styleUrl: './todo.css'
})
export class Todo {

  constructor(private http: HttpClient) {
    this.loadTodos();
  }

  // Backend API URL
  apiUrl = 'http://localhost:5292/api/todo';

  todos: any[] = [];

  newTodo = {
    title: '',
    description: '',
    priority: 'Low',
    dueDate: '',
    completed: false
  };

  // 🔹 GET ALL TODOS FROM BACKEND
  loadTodos() {
    this.http.get<any[]>(this.apiUrl).subscribe(data => {
      this.todos = data;
    });
  }

  // 🔹 ADD TODO TO DATABASE
  addTodo() {
    this.http.post(this.apiUrl, this.newTodo)
      .subscribe(response => {
        console.log('Saved:', response);
        this.loadTodos(); // refresh list
      });

    this.newTodo = {
      title: '',
      description: '',
      priority: 'Low',
      dueDate: '',
      completed: false
    };
  }

  // 🔹 DELETE TODO FROM DATABASE
  deleteTodo(id: number) {
    this.http.delete(`${this.apiUrl}/${id}`)
      .subscribe(() => {
        this.loadTodos();
      });
  }

  // 🔹 UPDATE TODO IN DATABASE
  updateTodo(todo: any) {
    this.http.put(`${this.apiUrl}/${todo.id}`, todo)
      .subscribe(() => {
        console.log('Updated');
      });
  }
}